runtime/src/iree/hal/drivers/hip/pending_queue_actions.c

```diff
@@ -198,6 +196,21 @@ IREE_TYPED_ATOMIC_SLIST_WRAPPER(iree_hal_hip_ready_action,
                                 offsetof(iree_hal_hip_atomic_slist_entry_t,
                                          slist_next));
 
+// Ready action atomic slist entry struct.
+typedef struct iree_hal_hip_atomic_slist_completion_t {
+  void (*callback)(void* user_data);
```

Comment:
The callback has `void(void*)` limitation due to the signature of launch host func. Now we are not doing that it should be able to return some error? I guess we are sort a bit loose on error handling overall which need to be fixed consistently. So it's fine if you want to put just a TODO here and fix it later.